xwiki-platform-core/xwiki-platform-test/xwiki-platform-test-ui/src/main/java/org/xwiki/test/ui/po/FormContainerElement.java

```diff
@@ -73,20 +73,25 @@ protected WebElement getFormElement()
 
     public void fillFieldsByName(Map<String, String> valuesByNames)
     {
-        Map<WebElement, String> valuesByElements = new LinkedHashMap<>((int) (valuesByNames.size() / 0.75));
-
+        Map<WebElement, String> valuesByElements = new LinkedHashMap<>();
+        
+        WebElement lastElement = null;
         for (String name : valuesByNames.keySet()) {
-            valuesByElements.put(getFormElement().findElement(By.name(name)), valuesByNames.get(name));
+            lastElement = getFormElement().findElement(By.name(name));
+            valuesByElements.put(lastElement, valuesByNames.get(name));
         }
         fillFieldsByElements(valuesByElements);
         
         /* Register password confirmation is usually the last element that needs to be validated by liveValidation.
           This wait allows to solve a race condition between the form submission and the computation of the status of
           those fields. We force the status to be solved before we try anything else, especially submitting the form.
+          Unfortunately in Java17 we do not have lastEntry() from LinkedHashMaps, 
+          so we use a few non optimized operations instead. 
+          This is okay because the Map should not contain a lot of elements.
           */
-        if (valuesByNames.containsKey("register2_password")) {
-            getDriver().waitUntilCondition(driver -> !getFormElement().findElement(By.name("register2_password"))
-                .getAttribute(CLASS_ATTRIBUTE).isEmpty());
+        if(!valuesByElements.isEmpty()) {
+            WebElement finalLastElement = lastElement;
```

Comment:
you don't need that variable, do you?

Reply:
The code was underlined with `Variable used in lambda expression should be final or effectively final` .

Reply:
Ah right, you use it in a lambda.